pkg/systemd/terminal.jsx

```diff
@@ -40,8 +40,27 @@
 
         constructor(props) {
             super(props);
-            const theme = document.cookie.replace(/(?:(?:^|.*;\s*)theme_cookie\s*=\s*([^;]*).*$)|^.*$/, "$1");
-            const size = document.cookie.replace(/(?:(?:^|.*;\s*)size_cookie\s*=\s*([^;]*).*$)|^.*$/, "$1");
+
+            let theme = localStorage.getItem('terminal:theme');
+            let size = localStorage.getItem('terminal:font-size');
+            // HACK: Try to read the configuration from localStorage, if it does not exists fall back
+            // to the old configuration stored in a browser's cookie. After enough time has been
+            // passed this can be dropped.
+            if (theme === null || theme === "") {
+                theme = document.cookie.replace(/(?:(?:^|.*;\s*)theme_cookie\s*=\s*([^;]*).*$)|^.*$/, "$1");
+                if (theme !== "") {
+                    localStorage.setItem('terminal:theme', theme);
+                    this.invalidateCookie("theme_cookie");
+                }
+            }
+            if (size === null || size === "") {
+                size = document.cookie.replace(/(?:(?:^|.*;\s*)size_cookie\s*=\s*([^;]*).*$)|^.*$/, "$1");
+                if (size !== "") {
+                    localStorage.setItem('terminal:font-size', size);
+                    this.invalidateCookie("size_cookie");
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18499-20230314-165143-eda2a345-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)